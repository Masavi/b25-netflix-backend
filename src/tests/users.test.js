// Imports: GraphQL & Schema
const { graphql } =  require('graphql')
const { makeExecutableSchema } = require('graphql-tools')

// Imports: GraphQL TypeDefs & Resolvers
import TypeDefs from '../graphql/typedefs/types';
import Query from '../graphql/resolvers/query';
import Mutation from '../graphql/resolvers/mutation';

const schema = makeExecutableSchema({
    typeDefs: TypeDefs,
    resolvers: { Query, Mutation },
});

import User from '../models/User';
import setupTest from './helpers';

const mutationRegister = `

    mutation Register($first_name:String!,$last_name:String!,$email:String!,
        $password:String!){
        signup(data:{first_name:$first_name,
            last_name:$last_name,
            email:$email,
            password:$password
        }){
            token
        }
    }
`

const queryMe = `

    query Profile{
        me{
            first_name,
            last_name,
            email
        }
    }

`

describe("Register user works correctly", () => {

    beforeEach( async () => await setupTest())

    it("probando suma de 2 + 2 = 4...", () => {
        return expect(2+2).toBe(4);
    });


     it("Should create user correctly",async () => {
        const first_name = "Prueba"
        const last_name = "Prueba"
        const email = "prueba@email.com"
        const password = "miprueba"

        const res = await graphql(
                                schema,
                                mutationRegister,
                                null,
                                {}
                                ,{first_name,last_name,email,password})

        expect(res).toHaveProperty("data")
        expect(res.data).toHaveProperty("signup")
        expect(res.data.signup).toHaveProperty("token")
    })

    it("Should retrieve me profile",async () => {

        const me =  await User.create({first_name:"prueba",last_name:"prueba",email:"prueba@prueba.com",password:"prueba"})

        const res  =  await graphql(schema,queryMe,null,{user:me},{})
        
        expect(res.data.me.first_name).toBe(me.first_name)

    })
})