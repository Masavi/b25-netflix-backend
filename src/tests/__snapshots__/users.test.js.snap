// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register user works correctly Should create user correctly 1`] = `
Object {
  "data": Object {
    "signup": Object {
      "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjViZmRjZGUyOWRkZjljMDVkMDg5NmE5NyIsImVtYWlsIjoicHJ1ZWJhQGVtYWlsLmNvbSIsImZpcnN0X25hbWUiOiJQcnVlYmEiLCJsYXN0X25hbWUiOiJQcnVlYmEiLCJpYXQiOjE1NDMzNTk5NzB9.8Gsgu1wPi36hQglkbgVx3qScuJkFw34LrZEqcszQFy0",
    },
  },
}
`;

exports[`Register user works correctly Should retrieve me profile 1`] = `
Object {
  "data": Object {
    "me": Object {
      "email": "prueba@prueba.com",
      "first_name": "prueba",
      "last_name": "prueba",
    },
  },
}
`;
